src/auto4_lua_dialog.cpp

```diff
@@ -328,6 +328,12 @@ namespace Automation4 {
 			{
 				lua_getfield(L, -1, "items");
 				read_string_array(L, items);
+
+#ifdef __WXMAC__
```

Comment:
Why is this mac-specific? Is there an upstream bug?

Reply:
The issue was with dropdowns whose `value` is the empty string (and that empty string is not part of the `items`). The automation docs for `value` say
> Determines which item is selected when the dialog is first displayed. If this is not one of the items specified, no item is selected. This is case- sensitive.

This works correctly on Windows and Linux, but didn't on Mac, where such dropdowns would just preselect one of the actual `items`. This happened with DependencyControl's install dialog, for example.

The wx docs for (the equivalent of) `value` say
> Initial selection string. An empty string indicates no selection.

so this does indeed seem to be an upstream bug. I guess this code could also run unconditionally but at the time I wanted to be minimally invasive with this change.

Reply:
not sure about wx issue trackers, but this has been a decently-long observed behavior of the depctrl install dialog

Reply:
I didn't find a report about it so I opened one now: https://github.com/wxWidgets/wxWidgets/issues/25011
I also added a comment.